zkevm-circuits/src/evm_circuit/util/memory_gadget.rs

```diff
@@ -61,6 +69,32 @@ pub(crate) mod address_high {
     }
 }
 
+/// Memory address trait to adapt for right and Uint overflow cases.
+pub(crate) trait CommonMemoryAddressGadget<F: Field> {
```

Comment:
Is the CommonMemoryAddessGadget trait necessary?
We only use MemoryAddressGadget for the no overflow case, and we only use MemoryExpandedAddressGadget for Sha3 oog when the overflow is possible.
In the current PR it seems the traits are not super useful as we already know the concrete struct in the function body that consumes it.

I don't plan to block the PR longer, but it would be great if you could share some of the rationale for adding the trait.

Reply:
the rationale is to design common methods for normal and error memory address gadget use , they both have  `offset`, `length` etc.  we can see successful opcode also use it.  @ChihChengLiang  does it make sense?

Reply:
it is also used in oog static/dynamic memory error cases etc.cases

Reply:
Thanks for sharing the rationale. I think it makes sense to me.